dashboard/http_server_head.py

```diff
@@ -128,6 +129,17 @@ def get_address(self):
         assert self.http_host and self.http_port
         return self.http_host, self.http_port
 
+    @aiohttp.web.middleware
+    async def path_clean_middleware(self, request, handler):
+        if request.path.startswith("/static") or request.path.startswith("/logs"):
+            parent = "/logs" if request.path.startswith("/logs") else "/static"
+
+            # If the destination is not relative to the expected directoy, then the user is attempting
+            # path traversal, so deny the request
+            if not pathlib.Path(os.path.abspath(request.path)).is_relative_to(parent):
```

Comment:
```suggestion
            if not pathlib.Path(request.path).absolute().is_relative_to(parent):
```

Mixing path libraries for a particular reason? (If its load bearing can you explain it in a comment?)

Reply:
Switched to `resolve()`